src/compiler/builder.ts

```diff
@@ -30,10 +30,18 @@ namespace ts {
          * These will be commited whenever the iteration through affected files of current changed file is complete
          */
         currentAffectedFilesSignatures: Map<string> | undefined;
+        /**
+         * Newly computed visible to outside referencedSet
+         */
+        currentAffectedFilesExportedModulesMap: BuilderState.ComputingExportedModulesMap | undefined;
         /**
          * Already seen affected files
          */
         seenAffectedFiles: Map<true> | undefined;
+        /**
+         * True if the semantic diagnostics were copied from the old state
+         */
+        semanticDiagnosticsFromOldStateFiles?: number;

```

Comment:
This `number` isn't used except as a counting semaphore (it only increments and decrements) and it has a lot of debug asserts around that - this leads me to believe it should more naturally be a `Map<true>` instead. (And then truthiness checks would be on `semanticDiagnosticsFromOldStateFiles.size` instead) Although changing the type of `semanticDiagnosticsPerFile`  from `Map<ReadonlyArray<Diagnostic>>` to `Map<{ fromOldState: boolean, diagnostics: ReadonlyArray<Diagnostic> }>` could be even better and make even more of the sanity checking assertions unnecessary.